scene/3d/visibility_notifier_3d.cpp

```diff
@@ -80,13 +80,16 @@ AABB VisibilityNotifier3D::get_aabb() const {
 void VisibilityNotifier3D::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_ENTER_WORLD: {
-			get_world_3d()->_register_notifier(this, get_global_transform().xform(aabb));
+			world = get_world_3d();
```

Comment:
What's the advantage of caching it, as opposed to just assigning to a local variable and doing `ERR_FAIL_COND` check? The same check might be need in `NOTIFICATION_TRANSFORM_CHANGED` too, or you might end up operating on a different `world` than the one cached maybe?

Reply:
It guarantees that on EXIT_WORLD you exit the same world you entered. The crash happened because for some reason notifier was losing the world reference. If you add only null check, the old world stays "unexited" and crashes on exit.

And yeah, replacing other references might be a good idea.